compiler/test/compilable/test18151.d

```diff
@@ -0,0 +1,30 @@
+// https://issues.dlang.org/show_bug.cgi?id=18151
+
+/*
+TEST_OUTPUT:
+---
+value
+---
+*/
+
+void test()(auto ref Inner inner)
+{
+    pragma(msg, __traits(isRef, inner) ? "ref" : "value");
+}
+
+struct Inner
+{
+}
+
+struct Outer
+{
+    Inner inner;
+    Inner get() { return inner; }
```

Comment:
Can you add another Outer struct that which returns `Inner` by ref, to make the test complete?